shap/_explanation.py

```diff
@@ -158,9 +161,10 @@ def __init__(
         )
 
     @property
-    def shape(self):
+    def shape(self) -> tuple[Union[int, None], ...]:
```

Comment:
I am a bit unsure about this one: should it be a tuple of `int`, or a tuple or `int|None` ?

The function `_compute_shape` seems to return a tuple that may contain `None`; however, the attribute `.shape` seems to be used in contexts that always expect an integer.

Reply:
Two things here: we cannot use the pipe like `int | None` yet, since we still support Python 3.9 (this syntax is only available in Python 3.10).

Yes, shape should return an integer but if it doesn't return this as of now, then I would argue that we just add a todo in this PR and do not confuse PRs that solely add type annotations with fixing issues.

Reply:
> we cannot use the pipe

Yep, that was just my shorthand in my comment above, but in the PR I've used the full `Union` syntax.

> shape should return an integer

Presume you mean a tuple of integers. Are you sure this should always be the case? I noticed on lines 788 & 816 that entries may be `None` if the array is sparse.

Reply:
I am not sure tbh. So I wouldn't change this but just stick to what is implemented as of now.

Then in a later iteration we can check suspicious type hints and go over those pieces of the code.

Reply:
OK - I'll add a TODO comment to remind myself to come back to this one. For now, I've used an `assert` on lines 108 and 639 to validate that the shape is not None